src/main/java/ch/njol/skript/expressions/ExprStatistics.java

```diff
@@ -0,0 +1,234 @@
+		ofType = exprs[matchedPattern + 1];
+		players = (Expression<OfflinePlayer>) exprs[matchedPattern == 0 ? 2 : 0];
+		return true;
+	}
+
+	@Override
+	@Nullable
+	public Long[] get(Event event) {
+		OfflinePlayer[] players = this.players.getArray(event);
+		Statistic[] statistics = this.statistics.getArray(event);
+		Object ofType = this.ofType != null ? this.ofType.getSingle(event) : null; // TODO support getArray()
+
+		if (players == null || statistics == null)
+			return new Long[0];
+
+		List<Long> result = new ArrayList<>(players.length * statistics.length);
+
+		for (OfflinePlayer player : players) {
+			for (Statistic statistic : statistics) {
+				try {
+					if (ofType instanceof ItemType) {
+						result.add((long) player.getStatistic(statistic, ((ItemType) ofType).getMaterial()));
+					} else if (ofType instanceof EntityData<?>) {
+						result.add((long) player.getStatistic(statistic, EntityUtils.toBukkitEntityType((EntityData<?>) ofType)));
+					} else {
+						result.add((long) player.getStatistic(statistic));
+					}
+				} catch (IllegalArgumentException ignored) {
+					return new Long[0];
+				}
+			}
+		}
+		return result.toArray(new Long[0]);
+	}
+
+	@Override
+	public boolean isSingle() {
+		return statistics.isSingle() && players.isSingle();
+	}
+
+	@Override
+	public Class<? extends Long> getReturnType() {
+		return Long.class;
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case SET:
+			case REMOVE:
+			case ADD:
+			case RESET:
+			case DELETE:
+				return CollectionUtils.array(Number.class);
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		OfflinePlayer[] players = this.players.getArray(event);
+		Statistic[] statistics = this.statistics.getArray(event);
+		Object ofType = this.ofType != null ? this.ofType.getSingle(event) : null; // TODO support getArray()
+
+		if (ofType instanceof ItemType) {
+			ofType = ((ItemType) ofType).getMaterial();
+		} else if (ofType instanceof EntityData) {
+			ofType = EntityUtils.toBukkitEntityType((EntityData<?>) ofType);
+		}
+
+		int value = (mode == ChangeMode.RESET || mode == ChangeMode.DELETE) ? 0 : ((Long) delta[0]).intValue();
+
+		applyStatistic(players, statistics, ofType, value, mode);
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "statistic " + statistics.toString(event, debug) + (ofType != null ? " of " +
+			players.toString(event, debug) : "") + " of " + players.toString(event, debug);
+	}
+
+	private static void applyStatistic(OfflinePlayer[] players, Statistic[] statistics, @Nullable Object ofType, int value, ChangeMode mode) {
+		for (OfflinePlayer p : players) {
+			for (Statistic statistic : statistics) {
+				try {
+					applyStatistic(p, statistic, ofType, value, mode);
+				} catch (IllegalArgumentException ex) {
+					return;
```

Comment:
Would this be better as a `continue`? If the player provides two statistics and no ofType, but 1 requires a type, then wouldn't it be better to at least change the one that can be changed instead of changing neither (or just the statistics before the problematic one?)

Reply:
Good catch!